src/IoTSettings.ts

```diff
@@ -59,21 +48,12 @@ export class IoTWorkbenchSettings {
     return _workbenchPath;
   }
 
-  async workbenchPath() {
-    const userWorkbenchPath = ConfigHandler.get<string>('workbench');
-    if (userWorkbenchPath) {
-      return userWorkbenchPath;
-    } else {
-      // Use the default value for workbenchPath.
-      await ConfigHandler.update(
-          'workbench', this._workbenchPath, vscode.ConfigurationTarget.Global);
-      return this._workbenchPath;
-    }
+  getWorkbenchPath(): string {
+    return ConfigHandler.get<string>(ConfigKey.workbench) || this.workbenchPath;
   }
 
-  async setWorkbenchPath(showMessage = true) {
-    let userWorkbenchPath: string|undefined =
-        ConfigHandler.get<string>('workbench') || this._workbenchPath;
+  async setWorkbenchPath(showMessage = true): Promise<void> {

```

Comment:
In which case, we would call this function with showMessage=false? I am not sure if the parameter is necessary.

Reply:
Will remove it since currently no scenario will set `showMessage` as false.